core/bones/base.py

```diff
@@ -882,3 +882,36 @@ def iter_bone_value(
                     yield idx, None, val
             else:
                 yield None, None, value
+
+    def structure(self) -> dict:
+        """
+        Describes the bone and its settings as an JSON-serializable dict.
+        This function has to be implemented for subsequent, specialized bone types.
+        """
+        ret = {
+            "descr": str(self.descr),  # need to turn possible translate-object into string
+            "type": self.type,
+            "required": self.required,
+            "params": self.params,
+            "visible": self.visible,
+            "readonly": self.readOnly,
+            "unique": self.unique.method.value if self.unique else False,
+            "languages": self.languages,
+            "emptyvalue": self.getEmptyValue(),  # fixme: rename this into "emptyvalue", same as "defaultvalue"
```

Comment:
The #fixme is now obsolete!
```suggestion
            "emptyvalue": self.getEmptyValue(),
```